src/pages/academy/gameSimulator/subcomponents/chapterPublisher/ChapterPublisher.tsx

```diff
@@ -0,0 +1,42 @@
+import { memo, useState } from 'react';
+import { useRequest } from 'src/commons/utils/Hooks';
+import { defaultChapter } from 'src/features/gameSimulator/GameSimulatorConstants';
+import { fetchChapters, fetchTextAssets } from 'src/features/gameSimulator/GameSimulatorService';
+import { ChapterDetail } from 'src/features/gameSimulator/GameSimulatorTypes';
+
+import ChapterPublisherEditor from './ChapterPublisherEditor';
+
+/**
+ * This components renders the Chapter Publisher component in the Game Simulator.
+ *
+ * @param textAssets - List of all text assets on S3 to choose from.
+ */
+const ChapterPublisher = memo(() => {
+  const { value: textAssets } = useRequest<string[]>(fetchTextAssets, []);
+  const { value: chapters } = useRequest<ChapterDetail[]>(fetchChapters, []);
+
+  const [chosenIndex, setChosenIndex] = useState(-1);
+
+  return (
+    <>
+      <h3>Publish / Edit Chapters</h3>
+      <select className="bp4-menu" onChange={(e: any) => setChosenIndex(e.target.key)}>
```

Comment:
We should be using `bp5-menu` since the CSS namespace is now Blueprint 5.

Reply:
Addressed in [d6b805d](https://github.com/source-academy/frontend/pull/2836/commits/d6b805d1a82065dea06d44dae8eff4438b1abcf7).